#! /bin/sh /usr/share/dpatch/dpatch-run
## 18_vdr-1.3.26-pagedown-fix.dpatch by Udo Richter <udo_richter@gmx.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix handling of the "page down" function

@DPATCH@
diff -urNad vdr-1.3.26/osdbase.c /tmp/dpep.Vp2fkN/vdr-1.3.26/osdbase.c
--- vdr-1.3.26/osdbase.c	2005-06-12 12:44:22.000000000 +0200
+++ /tmp/dpep.Vp2fkN/vdr-1.3.26/osdbase.c	2005-06-18 16:39:23.488592296 +0200
@@ -350,6 +350,10 @@
   current += displayMenuItems;
   first += displayMenuItems;
   int last = Count() - 1;
+  if (first + displayMenuItems > last) {
+     first = last - displayMenuItems + 1;
+     if (first < 0) first = 0;
+     }
   int tmpCurrent = current;
   while (!SelectableItem(tmpCurrent) && ++tmpCurrent <= last)
         ;
